src/utils/__tests__/cosmosdb_model_composed_versioned.test.ts

```diff
@@ -0,0 +1,265 @@
+      });
+
+      const model = new MyComposedModel(container);
+      const result = await model.upsert(document)();
+
+      expect(containerMock.items.create).toHaveBeenCalledWith(
+        {
+          ...document,
+          id: documentId(document[aModelExternalKeyId], document[aModelPartitionField], expectedVersion),
+          version: expectedVersion
+        },
+        { disableAutomaticIdGeneration: true }
+      );
+      expect(E.isRight(result));
+      if (E.isRight(result)) {
+        expect(result.right).toEqual({
+          ...document,
+          ...someMetadata,
+          id: documentId(document[aModelExternalKeyId], document[aModelPartitionField], expectedVersion),
+          version: expectedVersion
+        });
+      }
+    }
+  );
+
+  it("should fail on query error when retrieving last version", async () => {
+    containerMock.items.query.mockReturnValueOnce({
+      fetchAll: () => Promise.reject(errorResponse)
+    });
+    const model = new MyComposedModel(container);
+
+    const result = await model.upsert(aMyDocument)();
+    expect(E.isLeft(result));
+    if (E.isLeft(result)) {
+      expect(result.left.kind).toBe("COSMOS_ERROR_RESPONSE");
+      if (result.left.kind === "COSMOS_ERROR_RESPONSE") {
+        expect(result.left.error.code).toBe(500);
+      }
+    }
+  });
+
+  it("should fail on query error when creating next version", async () => {
+    containerMock.items.query.mockReturnValueOnce({
+      fetchAll: () => Promise.resolve(new FeedResponse([], {}, false))
+    });
+    containerMock.items.create.mockRejectedValueOnce(errorResponse);
+    const model = new MyComposedModel(container);
+
+    const result = await model.upsert(aMyDocument)();
+    expect(E.isLeft(result));
+    if (E.isLeft(result)) {
+      expect(result.left.kind).toBe("COSMOS_ERROR_RESPONSE");
+      if (result.left.kind === "COSMOS_ERROR_RESPONSE") {
+        expect(result.left.error.code).toBe(500);
+      }
+    }
+  });
+});
+
+describe("update", () => {
+  it("should create a new document with explicit version", async () => {
+    const expectedNextVersion = 0;
+    const externalKey = aMyDocument[aModelExternalKeyId];
+    const pk = aMyDocument[aModelPartitionField];
+    const model = new MyComposedModel(container);
+
+    const result = await model.create(aMyDocument)();
+
+    expect(containerMock.items.create).toHaveBeenCalledWith(
+      {
+        ...aMyDocument, // base type
+        id: documentId(externalKey, pk, expectedNextVersion),
+        version: expectedNextVersion
+      },
+      { disableAutomaticIdGeneration: true }
+    );
+    expect(E.isRight(result));
+    if (E.isRight(result)) {
+      expect(result.right).toEqual({
+        ...aMyDocument,
+        ...someMetadata,
+        id: documentId(externalKey, pk, expectedNextVersion),
+        version: expectedNextVersion
+      });
+    }
+  });
+});
+
+describe("create", () => {
+  it("should create a new document with version 0", async () => {
```

Comment:
This title seems wrong. From the test I understand that you are "updating" a retrieved document, and this should create a new version as defined in row 193.

Reply:
Yes, the titles of the unit test was not corresponding with the scenario. Fixed in 8532fec
Thanks